services/web/server/src/simcore_service_webserver/conversations/_conversation_message_repository.py

```diff
@@ -0,0 +1,179 @@
+import logging
+from typing import cast
+
+from aiohttp import web
+from models_library.conversations import (
+    ConversationID,
+    ConversationMessageDB,
+    ConversationMessageID,
+    ConversationMessagePatchDB,
+    ConversationMessageType,
+)
+from models_library.groups import GroupID
+from models_library.rest_ordering import OrderBy, OrderDirection
+from pydantic import NonNegativeInt
+from simcore_postgres_database.models.conversation_messages import conversation_messages
+from simcore_postgres_database.utils_repos import (
+    get_columns_from_db_model,
+    pass_or_acquire_connection,
+    transaction_context,
+)
+from sqlalchemy import asc, desc, func
+from sqlalchemy.ext.asyncio import AsyncConnection
+from sqlalchemy.sql import select
+
+from ..db.plugin import get_asyncpg_engine
+from .errors import ConversationMessageErrorNotFoundError
+
+_logger = logging.getLogger(__name__)
+
+
+_SELECTION_ARGS = get_columns_from_db_model(
+    conversation_messages, ConversationMessageDB
+)
+
+
+async def create(
+    app: web.Application,
```

Comment:
Q: Why is the app coming as argument in a repository? I find this a bit weird tbh. but ok.